src/main/java/address/model/AddressBook.java

```diff
@@ -22,6 +22,15 @@
     private List<Person> persons = new ArrayList<>();
     private List<Tag> tags = new ArrayList<>();
 
+    //TODO: remove this empty constructor
+    public AddressBook (){
```

Comment:
I don't think it's possible to remove empty constructor for any classes to be serialized. All serializers (JAXB, GSON and Jackson) need an empty constructor to be present for deserialization.

Jackson allows another way to bypass this, but many annotations must be added to the constructor with parameters which will clutter the constructor signature. [There is a way to bypass this as well but it requires changing the compiler option](https://manosnikolaidis.wordpress.com/2015/08/25/jackson-without-annotations/).